connect-location/src/main/java/com/ifttt/location/ConnectLocation.java

```diff
@@ -48,7 +48,9 @@ private ConnectLocation(GeofenceProvider geofenceProvider, ConnectionApiClient c
 
     @Override
     public void onStateChanged(ConnectButtonState currentState, ConnectButtonState previousState, Connection connection) {
-        geofenceProvider.updateGeofences(connection.features);
+        if (currentState == ConnectButtonState.Enabled) {
+            geofenceProvider.updateGeofences(connection.features);
+        }
```

Comment:
This does not account for disabled state, where we want to remove the geofences.

Reply:
sorry was checking individual commits, and check for initial or disabled state was added here.

Reply:
Just thought about this: I think we might also need to modify the GeofenceProvider to take in the `connection.status`. I don't know if a connection is disabled, whether or not we are still going to get the field values back. If that's the case (meaning that the `connection.features` is going to be the same for enabled and disabled connections), we should find a way to differentiate and make sure we unregister geofences if the connection is disabled.

Reply:
According to the comment here https://github.com/IFTTT/ConnectSDK-Android/blob/f4b3b353b587df1e7a9c1df0c74c5dee3b692fa4/connect-api/src/main/java/com/ifttt/connect/Feature.java#L34, `userFeatures` will be null for a disabled connection. 

The parsing logic in the `updateGeofences`, that eventually registers the geofence will work for a disabled connection as well if the above assumption is valid.

Reply:
The comment states that the field is going to be null if the user has not yet enabled it, or the API call was anonymous. I verified using a disabled connection and the response actually contained the user feature data.